### PR TITLE
Fix issue when adding image when no blocks are present

### PR DESCRIPTION
**Pull request by planarvoid**
### Fix
There is an issue that when there are no blocks in the selection and the user wants to add an image. In that case the selection is set to 0 and the image is added to the beginning of the editor. 

I've added unit tests to cover this functionality as well

### Test
1. Turn on `aztec.visualEditor.addMediaAfterBlocks()` in the `MainActivity`
2. Set `EXAMPLE` to something like `<h1>Test 1</h1>\nBody`
3. Run the app
4. Click at the end of `Body`
5. Add an image
6. Check it gets added after the `Body` and not at the beginning of the editor

### Review
@AmandaRiu 

Make sure strings will be translated:

- [x] If there are new strings that have to be translated, I have added them to the client's `strings.xml` as a part of the integration PR.

### CODE REVIEW

File: aztec/src/main/kotlin/org/wordpress/aztec/formatting/LineBlockFormatter.kt [hunk 1]
```diff
@@ -239,6 +238,12 @@ class LineBlockFormatter(editor: AztecText) : AztecFormatter(editor) {
                 position = spanEnd
             }
         }
+        if (position <= 0 && selectionEnd != 0) {
+            position = editableText.indexOf("\n", selectionEnd)
+        }
+        if (position <= 0 && selectionEnd != 0) {
+            position = editableText.length
+        }
```
Comment (AmandaRiu) [inline aztec/src/main/kotlin/org/wordpress/aztec/formatting/LineBlockFormatter.kt]: At first I thought the duplicate if-statements were a bug, but after some testing I realized really quickly why this is needed. I think we can simplify this a bit and also make it clear that this behavior is intentional:

```suggestion
        if (position <= 0 && selectionEnd != 0) {
            // If the text contains "\n" return that as the position, else set the position to the end of the text
            position = editableText.indexOf("\n", selectionEnd).takeIf { it >= 0 } ?: editableText.length
        }
```

**Review by AmandaRiu — APPROVED** [review]
@planarvoid One thing I did notice is that if nothing is selected then the image is added to the end of the last block span detected which seems fine until you have regular text following the block. For example, if I set the `EXAMPLE` text to `"<h1>Test 1</h1>\nMy Test"`, then run the app and just tap to add a photo  (without tapping inside the editor so `selectionEnd` is 0), then the image is added between the Heading and the body. This is an edge case I realize and I'm not sure what the fix would be in that case other than either defaulting the `position` to either 0 to add the image to the top of the document, or to `editableText.length` to return the position at the end of the document. I think the second option would keep everything in line with expected functionality for Aztec, but either way I'm not sure it's a big enough problem to address in this PR. 

I also added a code comment for your review. Everything works great so approved, but will wait to merge in case you are still online to look at this real quick.

File: aztec/src/main/kotlin/org/wordpress/aztec/formatting/LineBlockFormatter.kt (hunk 1 above)
Comment (planarvoid) [reply aztec/src/main/kotlin/org/wordpress/aztec/formatting/LineBlockFormatter.kt]: looks good, thanks for the suggestion 👍

**Comment by planarvoid** [comment]
thanks for the review @AmandaRiu 

> One thing I did notice is that if nothing is selected then the image is added to the end of the last block span detected which seems fine until you have regular text following the block. For example, if I set the EXAMPLE text to "<h1>Test 1</h1>\nMy Test", then run the app and just tap to add a photo (without tapping inside the editor so selectionEnd is 0), then the image is added between the Heading and the body. This is an edge case I realize and I'm not sure what the fix would be in that case other than either defaulting the position to either 0 to add the image to the top of the document, or to editableText.length to return the position at the end of the document. I think the second option would keep everything in line with expected functionality for Aztec, but either way I'm not sure it's a big enough problem to address in this PR.

Good catch 👍 , I've added an edgecase solution just for this usecase (if selection start and end are 0, insert at the beginning of the entry).

**Comment by AmandaRiu** [comment]
Changes look good! I'll merge as soon as CI has passed 👍🏻
